ouroboros-network/test/Ouroboros/Network/PeerSelection/Test.hs

```diff
@@ -913,7 +914,7 @@ initScript = newTVarM
 stepScript :: MonadSTM m => TVar m (Script a) -> m a
 stepScript scriptVar = atomically (stepScriptSTM scriptVar)
 
-stepScriptSTM :: MonadSTM m => TVar m (Script a) -> STM m a
+stepScriptSTM :: MonadSTMTx stm => TVar_ stm (Script a) -> stm a
```

Comment:
Required or just because we can? This leaks `MonadSTMTx` into app code which adds cognitive overhead.

Reply:
Same answer as above.